ls1intum/Artemis
# Add file upload exercises to exam mode

### Checklist

### Motivation and Context


The exam mode currently doesn't have the file-upload-exercise type.

### Description
In this PR we add the file-upload exercise to the exam mode. Instructors can now create file upload exercises in the exercise groups page.
![image](https://user-images.githubusercontent.com/33342534/107115585-f705ab80-686d-11eb-855d-05b91afb9254.png)
![image](https://user-images.githubusercontent.com/33342534/107115869-caeb2a00-686f-11eb-96a5-bfbfa0310a97.png)

Assessment works already too, but only for one correction round. Assessment in the second correction round will be added in the next PR.

### Steps for Testing


1. Create a new exam, add a new exercise group and add a file-upload-exercise.
2. Test the exam exercise first in a test run, and then with a real exam.
![image](https://user-images.githubusercontent.com/33342534/107158352-bac76d80-6989-11eb-9e66-12edc3a806cd.png)

3. upload a file, and click on upload. Now the uploaded file should be visible below, and you should be able to download it again.

4. upload a different file, and test if it still works.
5. hand in the exam
6. In the summary at the end you should be able to download the file again.
 <img src="https://user-images.githubusercontent.com/33342534/107526331-06f7f500-6bb8-11eb-8992-64ee7690be47.png" width="400">

### Test Coverage



file-upload-exam-submission.component.ts: 94.92%
